src/metatrain/experimental/phace/model.py

```diff
@@ -0,0 +1,678 @@
+import logging
+import warnings
+from typing import Any, Dict, List, Literal, Optional, Tuple
+
+import metatensor.torch
+import torch
+from metatensor.torch import Labels, TensorBlock, TensorMap
+from metatomic.torch import (
+    AtomisticModel,
+    ModelCapabilities,
+    ModelMetadata,
+    ModelOutput,
+    NeighborListOptions,
+    System,
+)
+
+from metatrain.utils.abc import ModelInterface
+from metatrain.utils.additive import ZBL, CompositionModel
+from metatrain.utils.data.dataset import DatasetInfo, TargetInfo
+from metatrain.utils.dtype import dtype_to_str
+from metatrain.utils.metadata import merge_metadata
+from metatrain.utils.scaler import Scaler
+from metatensor.torch.operations._add import _add_block_block
+
+from .modules.center_embedding import embed_centers, embed_centers_tensor_map
+from .modules.cg import get_cg_coefficients
+from .modules.cg_iterator import CGIterator
+from .modules.initial_features import get_initial_features
+from .modules.layers import EquivariantLastLayer, Identity, InvariantMLP
+from .modules.message_passing import EquivariantMessagePasser, InvariantMessagePasser
+from .modules.precomputations import Precomputer
+from .modules.tensor_product import (
+    TensorProduct,
+)
+from .utils import systems_to_batch
+
+
+warnings.filterwarnings(
+    "ignore",
+    category=UserWarning,
+    message=("The TorchScript type system doesn't support instance-level annotations"),
+)
+
+
+class PhACE(ModelInterface):
```

Comment:
Can you do

```python
from .documentation import ModelHypers

class PhACE(ModelInterface[ModelHypers]]):
    ...
    def __init__(self, hypers: ModelHypers, dataset_info: DatasetInfo)
```

so that mypy can check that there are no problems? And same for the trainer